### PR TITLE
Mikeo key reuse

### PR DESCRIPTION
**Pull request by ounsworth**

### CODE REVIEW

File: draft-ietf-lamps-pq-composite-kem.md [hunk 1]
```diff
@@ -880,6 +880,14 @@ EDNOTE to IANA: OIDs will need to be replaced in both the ASN.1 module and in {{
 
 # Security Considerations
 
+### Key Reuse {#sec-cons-key-reuse}
+
+When using single-algorithm cryptography, the best practice is to always generate fresh key material for each purpose, for example when renewing a certificate, or obtaining both a TLS and S/MIME certificate for the same device, however in practice key reuse in such scenarios is often not catastrophic to security and therefore often tolerated. With composite keys we have a much stricter security requirement. However this reasoning does not hold in the PQ / Tradational hybrid setting.
```
Comment (br-hale) [inline draft-ietf-lamps-pq-composite-kem.md]: Suggested edit:

When using single-algorithm cryptography, the best practice is to always generate fresh keying material for each purpose, for example when renewing a certificate, or obtaining both a TLS and S/MIME certificate for the same device, however in practice key reuse in such scenarios is not always catastrophic to security and therefore often tolerated.

File: draft-ietf-lamps-pq-composite-kem.md [hunk 2]
```diff
@@ -880,6 +880,14 @@ EDNOTE to IANA: OIDs will need to be replaced in both the ASN.1 module and in {{
...
+
+Within the broader context of PQ / Traditional hybrids, we need to consider new attack surfaces that arise due to the hybrid constructions and did not exist in single-algorithm contexts. One of these is key reuse where the component keys within a hybrid are also used by themselves within a single-algorithm context. For example, it might be tempting for an operator to take already-deployed RSA keys and add an ML-KEM key to them to form a hybrid. Within a hybrid signature context this leads to a class of attacks referred to as "stripping attacks" where one component signature can be extracted and presented as a single-algorithm signature. Hybrid KEMs using a contacenation-style KEM combiner, as is done in this document, do not have the analagous attack surface because even if an attacker is able to extract and decrypt one of the component ciphertexts, this will yield a different shared secret than the overall shared secret derived from the composite, so any subsequent symmetric cryptographic operations will fail. However there is still a risk of key reuse which relates to certificate revocation.
```
Comment (br-hale) [inline draft-ietf-lamps-pq-composite-kem.md]: Suggested edit:

"However there is still a risk of key reuse which relates to certificate revocation, as well as general key reuse security issues."

File: draft-ietf-lamps-pq-composite-kem.md [hunk 3]
```diff
@@ -228,7 +247,7 @@ informative:
...
+This document introduces a set of Key Encapsulation Mechanism (KEM) schemes that use pairs of cryptographic elements such as public keys and cipher texts to combine their security properties. These schemes effectively mitigate risks associated with the adoption of post-quantum cryptography and are fully compatible with existing X.509, PKIX, and CMS data structures and protocols. This document defines eleven specific pairwise combinations, namely ML-KEM Composite Schemes, that blend ML-KEM with traditional algorithms such as RSA-OAEP, ECDH, X25519, and X448. For use within CMS, this document is intended to be coupled with the CMS KEMRecipientInfo mechanism in {{I-D.housley-lamps-cms-kemri}}. These combinations are tailored to meet security best practices and regulatory requirements. Composite ML-KEM is applicable in any application that would otherwise use ML-KEM, but wants the protection against breaks or catastrophic bugs in ML-KEM.
```
Comment (janklaussner) [inline draft-ietf-lamps-pq-composite-kem.md]: ...eleven specific pairwise combinations...
count is currently nine, right?

File: draft-ietf-lamps-pq-composite-kem.md [hunk 4]
```diff
@@ -885,29 +917,47 @@ EDNOTE to IANA: OIDs will need to be replaced in both the ASN.1 module and in {{
...
+### Second pre-image resistance of componet KEMs {#sec-cons-ct-collision}
 
-The composite KEM design specified in this document, and especially that of the KEM combiner specified in {{sec-kem-combiner}} means that the overall composite KEM algorithm should be considered to have the security strength of the strongest of its component algorithms; ie as long as one component algorithm remains strong, then the overall composite algorithm remains strong.
+The notion of a second pre-image resistant KEM is defined in [X-Wing] being the property that it is computationally difficult to find two different ciphertexts `c != c'` that will decapsulate to the same shared secret under the same public key. For the purposes of a hybrid KEM combiner, this property means that given two composite ciphertexts `(c1, c2)` and `(c1', c2')`, we must obtain a unique overall shared secret so long as either `c1 != c1'` or `c2 != c2'` -- i.e. the overall composite KEM is second pre-image resistant, and therefore IND-CCA2 secure so, long as one of the component KEMs is.
```
Comment (janklaussner) [inline draft-ietf-lamps-pq-composite-kem.md]: [...]and therefore IND-CCA2 secure so, long as [...]
-> [...] secure, so long as [...]

File: draft-ietf-lamps-pq-composite-kem.md [hunk 5]
```diff
@@ -885,29 +917,47 @@ EDNOTE to IANA: OIDs will need to be replaced in both the ASN.1 module and in {{
...
+
+Upon receiving a new certificate enrollment request, many certification authorities will check if the requested public key has been previously revoked due to key compromise. Often a CA will perform this check by using the public key hash. Therefore, even if both components of a composite have been previously revoked, the CA may only check the hash of the combined composite key and not find the revocations. Therefore, it is RECOMMENDED to avoid key reuse and always generate fresh component keys for a new composite. It is also RECOMMENDED that CAs performing revocation checks on a composite key should also check both component keys independently.
+
+## Policy for Deprecated and Acceptable Algorithms
 
+Traditionally, a public key or certificate contains a single cryptographic algorithm. If and when an algorithm becomes deprecated (for example, RSA-512, or SHA1), it is obvious that the public keys or certificates using that algorithm are to be considered revoked.
```
Comment (janklaussner) [inline draft-ietf-lamps-pq-composite-kem.md]: is this statement true? Afaik its more complicated in the real world. Is there an RFC describing this?

**Review by janklaussner — CHANGES_REQUESTED** [review]

**Pull request merged**